### ISSUE TITLE
`sentry-cocoa` installedIntegrationsNames was mutated during setSDK info in client

### ISSUE THREAD

**Issue by xcocoa** [getsentry/sentry-react-native#2955]
**OS:**
- [ ]  Windows
- [yes]  MacOS
- [ ]  Linux

**Platform:**
- [yes]  iOS
- [ ]  Android

**SDK:**
- [yes]  `@sentry/react-native` (>= 1.0.0)
- [ ]  `react-native-sentry` (<= 0.43.2)

**SDK version:** "4.7.1"

**`react-native` version:** 0.0.0

**Are you using Expo?**
- [ ]  Yes
- [ ]  No

**Are you using sentry.io or on-premise?**
- [ ]  sentry.io (SaaS)
- [ ]  on-premise

**If you are using sentry.io, please post a link to your issue so we can take a look:**

[Link to issue]

**Configuration:**

(`@sentry/react-native`)
```
Sentry.init({
  dsn: 'https://...@sentry.io/...'
  // other options
});
```

or

(`react-native-sentry`)
```
Sentry.config(
  'https://...@sentry.io/...'
  // other options
 ).install();
```

---
**I have following issue:**

[Description]

**Steps to reproduce:**
- Step 1
- Step 2

**Actual result:**

```
Sentry Application threw exception NSGenericException: *** Collection <__NSArrayM: 0x28076e970> was mutated while being enumerated.3 Sentry 0x00000001055099f8 -[SentryClient setSdk:] + [SentryClient.m : 644]
```

**Expected result:**

[Expected result]

**Comment by krystofwoldrich** [comment]
Hi,
thank you for the message, unfortunately just from the exception message we are not able to reproduce the issue and therefore fix it. 

Could you describe what are you doing when the error occurs, what are the steps to reproduce the error, the stack trace, and what your `Sentry.init` look like?

**Comment by philipphofmann** [comment]
Yes, that's an error in the Cocoa SDK, but this should be fixed with https://github.com/getsentry/sentry-cocoa/pull/2398 released in [7.31.2](https://github.com/getsentry/sentry-cocoa/blob/main/CHANGELOG.md#7312), which went out in November 2022.

**Comment by krystofwoldrich** [comment]
@philipphofmann Thank you for the info.

@xcocoa please upgrade to RN SDK v[4.9.0](https://github.com/getsentry/sentry-react-native/releases/tag/4.9.0) or newer.
